mage_ai/frontend/oracle/components/FlyoutMenu/index.tsx

```diff
@@ -108,42 +112,72 @@ function FlyoutMenu({
     }
   }, [open]);
 
-  return (
+  let depth = 0;
+  const buildMenuEl = (items: FlyoutMenuItemType[], uuid: string, visible: boolean) => (
     <FlyoutMenuContainerStyle
       compact={compact}
       style={{
-        display: !open ? 'none' : null,
-        left: left || 0,
-        top: (height || 0) + topOffset,
+        display: !visible && !submenuVisible[uuid] ? 'none' : null,
+        left: depth === 0 ? left || 0 : (compact ? (depth * UNIT * 20) : (depth * UNIT * 34)),
```

Comment:
Use constants in the `mage-ai/mage_ai/frontend/oracle/components/FlyoutMenu/index.style.tsx` file for the UNIT values, and then import those constants here. The UNIT * 20 or 34 is very specific.